src/jitlayers.cpp

```diff
@@ -1123,6 +1127,7 @@ namespace {
                 {
                     auto stream = *jl_ExecutionEngine->get_dump_llvm_opt_stream();
                     if (stream) {
+                        jl_printf(stream, "%s", before_stats_ss.str().c_str());
```

Comment:
You'll need to check here that the before string isn't empty, to catch the case where this is enabled halfway through.
Maybe you could just add a separate boolean that starts out false and gets set to true inside the first block?